extensions/amp-story/1.0/semantic-render.js

```diff
@@ -0,0 +1,178 @@
+/**
+ * Copyright 2020 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {Services} from '../../../src/services';
+import {createElementWithAttributes} from '../../../src/dom';
+import {dev} from '../../../src/log';
+import {dict} from '../../../src/utils/object';
+import {includes, startsWith} from '../../../src/string';
+
+/**
+ * Renders the page description, and videos title/alt attributes in the page.
+ * @param {!./amp-story-page.AmpStoryPage} page
+ * @param {!Array<?Element>} videos
+ */
+export function renderPageDescription(page, videos) {
+  const descriptionElId = `i-amphtml-story-${page.element.id}-description`;
+  const descriptionEl = createElementWithAttributes(
+    page.win.document,
+    'div',
+    dict({
+      'class': 'i-amphtml-story-page-description',
+      'id': descriptionElId,
+    })
+  );
+
+  const addTagToDescriptionEl = (tagName, text) => {
+    if (!text) {
+      return;
+    }
+    const el = page.win.document.createElement(tagName);
+    el./* OK */ textContent = text;
+    descriptionEl.appendChild(el);
+  };
+
+  addTagToDescriptionEl('h2', page.element.getAttribute('title'));
+
+  const fetches = [];
+
+  videos.forEach(videoEl => {
+    addTagToDescriptionEl('p', videoEl.getAttribute('alt'));
+    addTagToDescriptionEl('p', videoEl.getAttribute('title'));
+    fetches.push(
+      fetchCaptions(page, videoEl).then(text => {
+        addTagToDescriptionEl('p', text);
+      })
+    );
+  });
+
+  Promise.all(fetches).then(() => {
```

Comment:
We should register this as both the success and error handler. If one or more captions file failed to load, we still want to render at least the tag/description.

Reply:
Obsolete in new code.